wordle-web/components/Keyboard.vue

```diff
@@ -1,31 +1,46 @@
 <template>
   <v-row
     class="justify-center"
-    v-for="(keyboardRow, rowIndex) in keyboardKeys"
+    v-for="(keyboardRow, rowIndex) in keyboardLetterRows"
     :key="rowIndex"
     dense
   >
     <v-col
       cols="auto"
-      v-for="key in keyboardRow"
-      :key="key"
+      v-for="letter in keyboardRow"
+      :key="letter.char"
       :class="'ml-1 px-0'"
     >
-      <v-btn color="grey">
-        {{ key }}
-      </v-btn>
+      <LetterResult :letter="letter" :clickable="true" />
     </v-col>
   </v-row>
 </template>
 
 <script setup lang="ts">
-import { useDisplay } from "vuetify";
+import { Game } from "~/scripts/game";
+import { Letter } from "~/scripts/letter";
 
-const display = useDisplay();
+const game: Game = inject("GAME")!;
 
-const keyboardKeys = [
-  ["Q", "W", "E", "R", "T", "Y", "U", "I", "O", "P"],
-  ["A", "S", "D", "F", "G", "H", "J", "K", "L"],
-  ["Z", "X", "C", "V", "B", "N", "M", "👈"],
-];
+const keyboardLetterRows = computed(() => {
+  let keyboardLetterRows: Letter[][] = [];
+
+  const keyboardRows = [
+    ["Q", "W", "E", "R", "T", "Y", "U", "I", "O", "P"],
+    ["A", "S", "D", "F", "G", "H", "J", "K", "L"],
+    ["Z", "X", "C", "V", "B", "N", "M", "👈"],
+  ];
+
+  // Loop through all 3 keyboard rows
+  for(let keyboardRow of keyboardRows){
+    let keyboardLetterRow: Letter[] = []
+    // For each key in each row, convert to a letter object
+    for(let key of keyboardRow){
+      keyboardLetterRow.push(game.guessedLetters.find((letter) => letter.char === key) ?? new Letter(key));
+    }
+    keyboardLetterRows.push(keyboardLetterRow);
+  }
```

Comment:
```suggestion
  for(const keyboardRow of keyboardRows){
    const keyboardLetterRow: Letter[] = []
    // For each key in each row, convert to a letter object
    for(const key of keyboardRow){
      keyboardLetterRow.push(game.guessedLetters.find((letter) => letter.char === key) ?? new Letter(key));
    }
    keyboardLetterRows.push(keyboardLetterRow);
  }
```

When declaring variables, if they are not expected to be re-assigned, it's preferred to use `const`... A good way to get in the habit of this until it becomes second nature would be to start everything off as `const` until you actually need to reassign it, this way it forces you to think about _why_ you're using `let` vs `const` when you need to go back and change it.